lib/opal/rewriters/logical_operator_assignment.rb

```diff
@@ -0,0 +1,199 @@
+require 'opal/rewriters/base'
+
+module Opal
+  module Rewriters
+    class LogicalOperatorAssignment < Base
+      class AbstractHandler < self
+        def self.get(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.set(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.process(lhs, rhs, root_type)
+          s(root_type,
+            get(lhs, rhs),
+            set(lhs, rhs))
+        end
+
+        def self.new_temp
+          @@counter ||= 0
+          @@counter += 1
+          :"$or_equals_lhs_recvr_tmp_#{@@counter}"
```

Comment:
I was implementing cvars in Opal, I trust them :smile: 
What do you think about extracting and reusing the code that generates temporary variables from the compiler? Something simialar to this code somewhere in `Opal::TemporaryVariable.next`? I just feel that I reinvent a wheel here. Moreover, this code can be used later in other rewriters.

We could pass an instance of compiler to a rewriter, but for me it looks like a bad design. AST rewriters exist outside of the compiler/parser. Rewriter is kind of a big stateless function that takes one ast returns another one.